### PR TITLE
{bio}[GCC/11.2.0] StringTie v2.2.1

### PR DESCRIPTION
**Pull request by SethosII**
(created using `eb --new-pr`)

The patch is needed because `StringTie` expects the linked libraries in a fixed position. Without it you will get an error like this:

```
/mnt/software/easybuild/20.04/software/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: cannot open ./htslib/xlibs/lib/liblzma.a: No such file or directory
cram/cram_io.c:1291: error: undefined reference to 'lzma_stream_buffer_bound'
cram/cram_io.c:1297: error: undefined reference to 'lzma_easy_buffer_encode'
cram/cram_io.c:1313: error: undefined reference to 'lzma_easy_decoder_memusage'
cram/cram_io.c:1313: error: undefined reference to 'lzma_stream_decoder'
cram/cram_io.c:1331: error: undefined reference to 'lzma_code'
cram/cram_io.c:1344: error: undefined reference to 'lzma_code'
cram/cram_io.c:1355: error: undefined reference to 'lzma_end'
cram/cram_io.c:1360: error: undefined reference to 'lzma_end'
cram/cram_io.c:1355: error: undefined reference to 'lzma_end'
collect2: error: ld returned 1 exit status
make: *** [Makefile:179: stringtie] Error 1
```

### CODE REVIEW

**Comment by SethosII** [comment]
Test report by @SethosII
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
ribnode012 - Linux Ubuntu 20.04, x86_64, Intel(R) Xeon(R) CPU E5-2640 v2 @ 2.00GHz, Python 3.8.10
See https://gist.github.com/440924286ee8aa123aef9f9bf465f88f for a full test report.

File: easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    'ce4eec532bbbe39af462be844afa6395ab38fa3418ef5fc2431e2d6194129527',  # stringtie-2.2.1.tar.gz
+    'b82dae4790ea9472fe9fe0578b6659afd8006640c5ea488661a4e222103bab6a',  # StringTie-2.2.1_fix-linking.patch
+]
+
+dependencies = [
+    ('bzip2', '1.0.8'),
+    ('libdeflate', '1.8'),
+    ('XZ', '5.2.5'),
+    ('zlib', '1.2.11'),
+]
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb:29]:
```suggestion
    ('HTSlib', '1.14'),
]
```

File: easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+toolchain = {'name': 'GCC', 'version': '11.2.0'}
+
+source_urls = ['https://github.com/gpertea/%(namelower)s/releases/download/v%(version)s']
+sources = [SOURCELOWER_TAR_GZ]
+patches = ["%(name)s-%(version)s_fix-linking.patch"]
+checksums = [
+    'ce4eec532bbbe39af462be844afa6395ab38fa3418ef5fc2431e2d6194129527',  # stringtie-2.2.1.tar.gz
+    'b82dae4790ea9472fe9fe0578b6659afd8006640c5ea488661a4e222103bab6a',  # StringTie-2.2.1_fix-linking.patch
+]
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb:22]:
```suggestion
checksums = ['ce4eec532bbbe39af462be844afa6395ab38fa3418ef5fc2431e2d6194129527']
```

File: easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+dependencies = [
+    ('bzip2', '1.0.8'),
+    ('libdeflate', '1.8'),
+    ('XZ', '5.2.5'),
+    ('zlib', '1.2.11'),
+]
+
+# the test script downloads some test data from the internet
+buildopts = 'release && ./run_tests.sh'
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb:32]:
```suggestion
buildopts = 'HTSLIB="$EBROOTHTSLIB/lib" LIBS="-lhts -lbz2 -llzma -ldeflate -lz $LIBS"'

# the test script downloads some test data from the internet
runtest = 'test'
testopts = buildopts
```

**Review by jfgrimm — CHANGES_REQUESTED** [review]
@SethosII I've proposed a few changes to override makefile variables in the build/test steps instead of using a patch file. It also avoids building the bundled HTSlib and dynamically links the one we provide.

**Comment by SethosII** [comment]
@jfgrimm Thank you very much for your suggestions! As of https://github.com/gpertea/stringtie/issues/362#issuecomment-1092989585 there are no custom modifications to `HTSlib` included there, so it should be fine to use the normal one, although the included on is version `1.12`.

**Comment by jfgrimm** [comment]
@boegelbot: please test @ generoso

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node130.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/76e059b707101b5fadc45a20ac593c5c for a full test report.

File: easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+sources = [SOURCELOWER_TAR_GZ]
+checksums = ['ce4eec532bbbe39af462be844afa6395ab38fa3418ef5fc2431e2d6194129527']
+
+dependencies = [
+    ('bzip2', '1.0.8'),
+    ('libdeflate', '1.8'),
+    ('XZ', '5.2.5'),
+    ('zlib', '1.2.11'),
+    ('HTSlib', '1.14'),
+]
```
Comment (jfgrimm) [inline easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb:26]: the `prepDE.py` requires `python2`, so we should probably include that as a dependency (and add the appropriate versionsuffix)

**Review by jfgrimm — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb (hunk 4 above)
Comment (SethosII) [reply easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb:26]: Should we use the bare or non-bare Python variant here?

Comment (jfgrimm) [reply easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb:26]: full, the bare dependency is used when python is only a build dependency (but not runtime)

Comment (SethosII) [reply easybuild/easyconfigs/s/StringTie/StringTie-2.2.1-GCC-11.2.0.eb:26]: Thanks for the clarification, I will change this.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
login2.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/6e6cd95715773b4b39f1b05e6932121b for a full test report.

**Comment by jfgrimm** [comment]
Going in, thanks @SethosII!

**Pull request merged**